multiqc/modules/malt/malt.py

```diff
@@ -0,0 +1,151 @@
+            info='performs alignment of metagenomic reads against a database of reference sequences (such as NR, GenBank or Silva) and produces a MEGAN RMA file as output.'
+        )
+
+        # Find and load Malt reports
+        self.malt_data = dict()
+
+        for f in self.find_log_files('malt', filehandles=True):
+            self.parse_logs(f)
+
+        if len(self.malt_data) == 0:
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.malt_data)))
+
+        self.malt_general_stats()
+        self.mappability_barplot()
+        self.taxonomic_assignation_barplot()
+
+    def parse_logs(self, f):
+        """Parses a Malt log file"""
+        reading = False
+        keys = [
+            "Total reads",
+            "Assig. Taxonomy",
+            "Num. of queries",
+            "Aligned queries",
+            "Num. alignments",
+        ]
+        for line in f['f']:
+            line = line.rstrip()
+            if line.startswith("+++++ Aligning file:") and reading == False:
+                reading = True
+                s_name = line.split()[-1]
+                s_name = self.clean_s_name(s_name, f['root'])
+                if s_name in self.malt_data:
+                    log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+                self.malt_data[s_name] = {}
+            elif reading:
+                for k in keys:
+                    if line.startswith(k):
+                        self.malt_data[s_name][k] = int(line.split()[-1].replace(",", ""))
+                        if k == 'Num. alignments':
+                            try:
+                                self.malt_data[s_name]['Non mapped'] = self.malt_data[s_name]["Num. of queries"] - self.malt_data[s_name]['Total reads']
+                                self.malt_data[s_name]['No Assig. Taxonomy'] = self.malt_data[s_name]["Total reads"] - self.malt_data[s_name]['Assig. Taxonomy']
+                                self.malt_data[s_name]['Mappability'] = (float(self.malt_data[s_name]['Total reads']) / float(self.malt_data[s_name]["Num. of queries"]))*100.0
+                                self.malt_data[s_name]['Taxonomic assignment success'] = (float(self.malt_data[s_name]['Assig. Taxonomy']) / float(self.malt_data[s_name]['Total reads']))*100.0
+                            except KeyError:
+                                pass
+                            reading = False
+
+    def mappability_barplot(self):
+        """Mappability barplot
+
+        Mappability = (Total reads / Num. of queries) * 100
+        """
+        cats = OrderedDict()
+        cats['Total reads'] = {'name': 'Mapped reads'}
+        cats['Non mapped'] = {'name': 'Non Mapped reads'}
+        config = {
+            'id': 'malt-mappability-plot',
+            'title': 'MALT: Metagenomic Mappability',
+            'ylab': 'Samples',
```

Comment:
**Question** - are these y-axis labels really meant to be `Samples` and not `Number of reads`? I didn't feel confident enough to change this one.

Reply:
I was also a bit confused. Since it's a "rotated barplot", does the y-axis becomes the x-axis ?
It also made more sense to me if xlab is the 'Samples'.

![Screenshot 2020-05-30 at 12 58 59](https://user-images.githubusercontent.com/8630837/83326574-6655e500-a275-11ea-9d39-a8e60374404f.png)

Reply:
Oh well, that'll be for 1.20 ;)

Reply:
Yes, it's not super clear and you're not the first to be caught out by this. It's because the HighCharts plotting code calls it the y-axis because bar plots are usually the other way up. As it's rotated it should be x as you say, but then the code would have been more confusing for me to write. I made this decision way back in the early days and kind of feel that it's too late to change now!